app/api/crud.py

```diff
@@ -67,19 +75,61 @@ async def get(
     if image_modal:
         params["image_modal"] = image_modal
 
-    for node_url in node_urls:
-        node_name = util.FEDERATION_NODES[node_url]
-        response = util.send_get_request(node_url + "query/", params)
+    tasks = [
+        util.send_get_request(node_url + "query/", params)
+        for node_url in node_urls
+    ]
+    responses = await asyncio.gather(*tasks, return_exceptions=True)
 
-        for result in response:
-            result["node_name"] = node_name
+    for node_url, response in zip(node_urls, responses):
+        node_name = util.FEDERATION_NODES[node_url]
+        if isinstance(response, HTTPException):
+            node_errors.append(
+                {"node_name": node_name, "error": response.detail}
+            )
+            warnings.warn(
+                f"Query to node {node_name} ({node_url}) did not succeed: {response.detail}"
+            )
+        else:
+            for result in response:
+                result["node_name"] = node_name
+            cross_node_results += response
```

Comment:
```suggestion
            cross_node_results.extend(response)
```
:cherries: https://docs.python.org/3/tutorial/datastructures.html ?